src/main/java/crazypants/enderio/conduit/item/filter/ItemFilterLimited.java

```diff
@@ -0,0 +1,253 @@
+        boolean damageMatched = false;
+        if (checkDamage && item.getItem().isDamageable()) {
+            damageMatched = fuzzyMode.compare(item);
+        }
+
+        if (isItemListEmpty()) {
+            if (checkDamage) {
+                return damageMatched;
+            } else {
+                return false;
+            }
+        }
+
+        boolean matched = false;
+
+        if (filter != null && item.getItem() == filter.getItem()) {
+            matched = true;
+
+            if (checkDamage && !damageMatched) {
+                matched = false;
+            } else if (matchMeta && item.getItemDamage() != filter.getItemDamage()) {
+                matched = false;
+            } else if (matchNBT && !isNBTMatch(item, filter)) {
+                matched = false;
+            }
+        }
+
+        if (!matched && useOreDict && isOreDicMatch(filter, item)) {
+            matched = true;
+        }
+        return matched;
+    }
+
+    public boolean isOreDicMatch(ItemStack filter, ItemStack item) {
+        int[] id_filter = OreDictionary.getOreIDs(filter);
+        if (id_filter.length == 0) {
+            return false;
+        }
+        int[] id_item = OreDictionary.getOreIDs(item);
+        if (id_item.length == 0) {
+            return false;
+        }
+        for (int id1 : id_filter) {
+            for (int id2 : id_item) {
+                if (id1 == id2) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public ItemStack decrStackSize(int fromSlot, int amount) {
+        if ((fromSlot < 0) || (fromSlot >= items.length)) {
+            return null;
+        }
+        oreIds.set(fromSlot, null);
+        ItemStack item = items[fromSlot];
+        items[fromSlot] = null;
+        return item.splitStack(amount);
+    }
+
+    @Override
+    public String toString() {
+        return "Limited" + super.toString();
+    }
+
+    @Override
+    public String getUnlocalizedName() {
+        return "gui.limited_item_filter";
+    }
+
+    class ItemFilterGhostSlot extends GhostSlot {
+
+        private final int slot;
+        private final Runnable cb;
+
+        private final int adjustOnShift;
+
+        ItemFilterGhostSlot(int slot, int x, int y, Runnable cb) {
+            this.x = x;
+            this.y = y;
+            this.slot = slot;
+            this.cb = cb;
+            this.displayStdOverlay = true;
+            this.stackSizeLimit = 64;
+            this.adjustOnShift = 10;
+        }
+
+        @Override
+        public void putStack(ItemStack stack) {
+            if (stack != null) {
+                stack = stack.copy();
+                System.out.println("StackSize" + stack.stackSize);
+                if (Keyboard.isKeyDown(Keyboard.KEY_LSHIFT) || Keyboard.isKeyDown(Keyboard.KEY_RSHIFT)) {
+                    if (Mouse.isButtonDown(0) || Mouse.getEventDWheel() == 1) {
```

Comment:
For compatibility (with lwjgl2, for example):
```suggestion
                    if (Mouse.isButtonDown(0) || Mouse.getEventDWheel() > 0) {
```

Reply:
Also if I'm not mistaken this class is client-only, right?

Reply:
Sorry I dont´t understand, my first time modding. So it wouldn´t work correctly on a server?